File: iOS/Star-Wars-Demo-APP/Star-Wars-Demo-APP/Data/Remote/DTO/VimeoDTOs.swift
import Foundation

struct VimeoSearchResponseDTO: Decodable {
    let data: [VimeoVideoDTO]
}

struct VimeoVideoDTO: Decodable {
    let uri: String
    let link: String
    let name: String
}

struct VimeoVideoPlaybackDTO: Decodable {
    struct PlayDTO: Decodable {
        struct HlsDTO: Decodable {
            let link: String?
        }

        struct ProgressiveDTO: Decodable {
            let link: String
            let height: Int?
        }

        let hls: HlsDTO?
        let progressive: [ProgressiveDTO]?
    }

    let play: PlayDTO?
}

File: android/app/src/main/java/com/dam/starwarsapp/data/remote/dto/vimeo/VimeoDtos.kt
package com.dam.starwarsapp.data.remote.dto.vimeo

import com.squareup.moshi.Json

data class VimeoSearchResponseDto(
    val data: List<VimeoVideoItemDto> = emptyList(),
)

data class VimeoVideoItemDto(
    val uri: String? = null,
    val link: String? = null,
    val name: String? = null,
)

data class VimeoVideoDetailsDto(
    val uri: String? = null,
    val link: String? = null,
    val name: String? = null,
    val play: VimeoPlayDto? = null,
)

data class VimeoPlayDto(
    val progressive: List<VimeoProgressiveDto> = emptyList(),
)

data class VimeoProgressiveDto(
    val link: String? = null,
    val height: Int? = null,
    val quality: String? = null,
    @Json(name = "mime") val mime: String? = null,
    val type: String? = null,
)
